apps/client/src/components/addToast/addToast.tsx

```diff
@@ -1,37 +1,89 @@
-import { useRef } from 'react';
+import React, { useState } from 'react';
 import styles from './addToast.module.css';
+import { Dialog } from '@mui/material';
+import { useGetUsersQuery, useAddMutation } from '../../store';
+import { User } from '../../types/user.types';
+
+interface AddToastProps {
+  setIsOpenAddToastModal: React.Dispatch<React.SetStateAction<boolean>>;
+  isOpenAddToastModal: boolean;
+}
+
+export const AddToast: React.FC<AddToastProps> = ({
+  isOpenAddToastModal,
+  setIsOpenAddToastModal,
+}) => {
+  const [user, setUser] = useState('');
+  const [date, setDate] = useState('');
+
+  const [addToast, result] = useAddMutation({
+    fixedCacheKey: 'add-toast-result',
+  });
+
+  const { data: users } = useGetUsersQuery();
+  const handleAddToast = async () => {
+    addToast({
+      date: date,
+      userId: user,
+      hasHappened: true,
+    })
+      .unwrap()
+      .then((result) => {
+        console.log('added toast successfully');
+      });
+    setIsOpenAddToastModal(false);
+  };
 
-const LoginMenu = () => {
-  const name = useRef<string>('');
-  const password = useRef<string>('');
   return (
-    <div className={styles.loginMenu}>
-      <div className={styles.loginTitle}> Login </div>
-      <form className={styles.loginForm}>
-        <div className={styles.form}>
-          <div className={styles.titleAndBox}>
-            <label htmlFor="">username</label>
-            <input
-              className={styles.inputBox}
-              onChange={(e) => (name.current = e.target.value)}
-            />
+    <Dialog
+      open={isOpenAddToastModal}
+      onClose={() => setIsOpenAddToastModal(false)}
+      maxWidth={false}
+    >
+      <div className={styles.addToastMenu}>
+        <div className={styles.addToastTitle}> להוסיף שתייה </div>
+        <div className={styles.addToastForm}>
+          <div className={styles.form}>
+            <div className={styles.titleAndBox}>
+              <label htmlFor="">username</label>

```

Comment:
`htmlFor` should be the same as select `name`. Please research the relatinship between input and label and their respective label and name attributes.